pkg/announce/announce.go

```diff
@@ -143,3 +155,36 @@ func create(workDir, subject, message string) error {
 
 	return nil
 }
+
+// getGoVersion runs kube-cross container and go version inside it.
+// We're running kube-cross container because it's not guaranteed that
+// k8s-cloud-builder container will be running the same Go version as
+// the kube-cross container used to build the release.
+func getGoVersion(tag string) (string, error) {
+	semver, err := util.TagStringToSemver(tag)
+	if err != nil {
+		return "", errors.Wrap(err, "parse version tag")
+	}
+
+	kubecrossBranches := []string{
+		fmt.Sprintf("release-%d.%d", semver.Major, semver.Minor),
+		git.DefaultBranch,
+	}
+
+	kubecrossVer, err := release.GetKubecrossVersion(kubecrossBranches...)
+	if err != nil {
+		return "", errors.Wrap(err, "get kubecross version")
+	}
+
+	kubecrossImg := fmt.Sprintf("k8s.gcr.io/build-image/kube-cross:%s", kubecrossVer)
+
+	res, err := command.New(
+		"docker", "run", "--rm", kubecrossImg, "go", "version",
+	).RunSilentSuccessOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "get go version")
+	}
+
+	versionRegex := regexp.MustCompile(`^?([0-9]+)(\.[0-9]+)?(\.[0-9]+)`)
+	return versionRegex.FindString(strings.TrimSpace(res.OutputTrimNL())), nil
```

Comment:
Just a nitty nit, I would log the go version we found, it is this code's info we want to record in the logs in case we need to go back to them

Reply:
can be left for a follow up though :)

Reply:
@puerco Fixed, PTAL. :slightly_smiling_face: